Source/UI/Activation/Services/QRCodeScanner.swift

```diff
@@ -48,20 +55,23 @@ public class QRCodeScanner: NSObject, QRCodeProvider, AVCaptureMetadataOutputObj
     
     public weak var delegate: QRCodeProviderDelegate?
     
-    public var isScannerStarted: Bool {
+    public var isScannerRunning: Bool {
         return captureSession?.isRunning ?? false
     }
     
+    /// Starts the scanner asynchronously
     public func startScanner() {
         prepareSession()
         guard let session = captureSession else { return }
-        captureSessionQueue.async {
+        captureSessionQueue.async { [weak self] in
             if session.isRunning == false {
+                self?.reported = false
```

Comment:
It's better to move this to main thread, before the async block is issued. We should also notice in the method documentation, that public start / stop can be called only from main thread.